libxrdp/xrdp_sec.c

```diff
@@ -778,20 +778,20 @@ xrdp_sec_process_logon_info(struct xrdp_sec *self, struct stream *s)
         return 1;
     }
 
-    if (unicode_in(s, len_domain, self->rdp_layer->client_info.domain, 255) != 0)
+    if (unicode_in(s, len_domain, 512, self->rdp_layer->client_info.domain, 255) != 0)
```

Comment:
I don't find any uniform name for a define. the function argument explains the value purpose well enough, even when not defined.
We have around 5 different fields with the same protocol defined length.
I don't think 5 different define will make the code more clear.